site/migration-guides/entrypoint-v7.md

```diff
@@ -0,0 +1,200 @@
+  signer,
+  transport: custom(client),
+  chain,
+  entryPoint: getEntryPoint(chain, { version: "0.7.0" }), // required for Entrypoint v0.7 [!code ++]
+});
+
+const smartAccountClient = createSmartAccountClientFromExisting({
+  client,
+});
+
+const uoStruct_v7: UserOperationStruct<"0.7.0"> =
+  await smartAccountClient.buildUserOperation({
+    uo: {
+      target: "0xdeadbeefdeadbeefdeadbeefdeadbeefdeadbeef",
+      data: "0x",
+      value: 10n,
+    },
+    account: account_v7,
+  });
+
+// uoStruct_v7: UserOperationStruct<"0.7.0"> = {
+//   sender: string;
+//   nonce: BigNumberish;
+//   initCode: BytesLike | "0x";  [!code --]
+//   factory?: string;  [!code ++]
+//   factoryData?: BytesLike;  [!code ++]
+//   callData: BytesLike;
+//   callGasLimit?: BigNumberish;
+//   verificationGasLimit?: BigNumberish;
+//   preVerificationGas?: BigNumberish;
+//   maxFeePerGas?: BigNumberish;
+//   maxPriorityFeePerGas?: BigNumberish;
+//   paymasterAndData: BytesLike | "0x";  [!code --]
+//   paymaster?: string;  [!code ++]
+//   paymasterVerificationGasLimit?: BigNumberish;  [!code ++]
+//   paymasterPostOpGasLimit?: BigNumberish;  [!code ++]
+//   paymasterData?: BytesLike;  [!code ++]
+//   signature: BytesLike;
+// }
+```
+
+### [`UserOperationOverrides`](/resources/types#useroperationoverrides)
+
+Similar to the updated `UserOperation` type from `Entrypoint` v0.6 to v0.7, `UserOperationOverrides` types are different from the two version operations.
+
+#### `paymasterAndData` field replaced with `paymaster` address and the `paymasterData` separated in addition to the 2 added `paymaster` operation gas limit fields
+
+```ts
+export type UserOperationOverrides<'0.7.0'> = {
+  ...
+  paymasterAndData?: EmptyHex;  // [!code --]
+  paymaster?: string;  // [!code ++]
+  paymasterData?: EmptyHex;  // [!code ++]
+  paymasterVerificationGasLimit?: BigNumberish | Multiplier;  // [!code ++]
+  paymasterPostOpGasLimit?: BigNumberish | Multiplier;  // [!code ++]
+}
+```
+
+::: tip Note
+Note that per [`ERC-4337`](https://eips.ethereum.org/EIPS/eip-4337#parameters), paymaster fields (`paymaster`, `paymasterData`, `paymasterValidationGasLimit`, `paymasterPostOpGasLimit`) either all exist, or none in the final `UserOperation` request sent to the connected Bundler.
+:::
+
+#### `stateOverride` field added to `UserOperationOverrides` for validating and estimating gas for transactions
+
+The state overrides during transaction calls (e.g., [`eth_call`](https://docs.alchemy.com/reference/eth-call) and [`eth_estimateGas`](https://docs.alchemy.com/reference/eth-estimategas)) allow for more flexible testing and gas estimation. However, not all networks support this feature. For networks lacking state overrides, `Entrypoint` v0.7 introduced the `delegateAndRevert()` function to be used as a workaround. It aids in validating and estimating gas for transactions by simulating conditions without permanently changing the on-chain state.
```

Comment:
As we discussed in person: this is not what `delegateAndRevert` is for. `delegateAndRevert` is useful internally for bundlers but not for common uses for most developers.

The new state overrides, as an argument to `eth_estimateUserOperationGas` are a separate feature and does depend on state overrides being present in the chain using them. This is because general-purpose state overrides can change the state of any address, including balance, code, or storage, while `delegateAndRevert` is only useful for changing the code of the entry point.

Reply:
<img width="774" alt="Screenshot 2024-04-19 at 9 42 34 PM" src="https://github.com/alchemyplatform/aa-sdk/assets/3278577/b9225524-77fc-4d80-b849-c154153467d6">
but according to the official v0.7 release note, and the linked [PR](https://github.com/eth-infinitism/account-abstraction/pull/365) for `delegateAndRevert`, it says 

"Minimal "static" call from EntryPoint to simulate special cases. Equivalent to the EntryPointSimulations in purpose, but doesn't require stateOverride.
Executes code on behalf of entryPoint using delegateCall, and revert with the result"

This sounds like this enables similar start overrides-like simulation to be done on chains without state overrides support in eth_call/eth_estimateGas?

cc: @dancoombs